### DISCUSSION TITLE
`Error: spawn sh ENOENT` when running `vercel build`

### DISCUSSION THREAD

**Discussion by hansottowirtz** [vercel/community#2793] [Help]
### Summary

When running `vercel build`:

```
❯ vercel build
Vercel CLI 30.2.1
? No Project Settings found locally. Run `vercel pull` for retrieving them? [
Y/n] y
? Set up “<redacted>”? [Y/n] y
? Which scope should contain your project? <redacted>
? Found project “<redacted>. Link to it? [Y/n] y
🔗  Linked to <redacted> (created .vercel)
> Downloading `preview` Environment Variables for Project <redacted>
✅  Created .vercel/.env.preview.local file [212ms]

> Downloading project settings
✅  Downloaded project settings to .vercel/project.json [0ms]
Error: spawn sh ENOENT
```

### Steps to Reproduce

`vercel build`

**Comment by amyegan** [comment]
Do you see more details if you run it with the [`--debug` option](https://vercel.com/docs/cli/global-options#debug)?

**Comment by hansottowirtz** [comment]
I've figured it out: it's because Vercel is running `sh` in a non-existent directory, and I'm using Vercel in a monorepo. 

`apps/web/vercel.json` (generated by the CLI):
```json
{
  "rootDirectory": "apps/web",
  ...
}
```

On [this line](https://github.com/vercel/vercel/blob/d61a1a79886f090c7f6fa06672115d6947d04cfc/packages/build-utils/src/fs/run-user-scripts.ts#L78), `opts.cwd` is `"<root>/apps/web/apps/web"`.

It can be solved by removing `rootDirectory` from `vercel.json` manually. But I think that `vercel.json` shouldn't have been generated with `rootDirectory` in this situation.

**Comment by hansottowirtz** [comment]
Very related: https://github.com/orgs/vercel/discussions/2845#discussioncomment-6169526
